### DISCUSSION TITLE
babel.transform() from es2017 to es2015 takes about 5 minutes

### DISCUSSION THREAD

**Discussion by gogo9th** [babel/babel#12318] [Q&A]
I am trying to transform this file from es2017 to es2015: 

[sample-beautify.js](https://drive.google.com/file/d/1RR0JaWKt-o0RJGldoQo8QrwfmmF4S4hB/view?usp=sharing)

But it takes about 5 minutes... My transformation code is this:

```
const babel = require('babel-core');
const fs = require('fs')

const code = fs.readFileSync('sample-beautify.js', 'utf8');
babel.transform(code, {retainLines: true, presets: ['es2017', 'es2015']});

```

Is there any way to make it faster by modifying my code? Any help will be greatly appreciated.

**Comment by gogo9th** [comment]
I am trying to run the above code with the latest babel library. I first upgraded the package as follows:

npm install @babel/core @babel/preset-es2015 @babel/preset-es2017

But how should I re-write my code to make it work with the updated latest babel library? For example, the latest babel library does not have 'babel-core'. 

```
const babel = require('babel-core');
const fs = require('fs')

const code = fs.readFileSync('sample-beautify.js', 'utf8');
babel.transform(code, {retainLines: true, presets: ['es2017', 'es2015']});

```

**Reply by existentialism** [reply]
Try `require('babel-core')` -> `require('@babel/core')`

**Reply by nicolo-ribaudo** [reply]
Also the `es2015` and `es2017` presets are now replaced by `@babel/preset-env`.

**Reply by gogo9th** [reply]
If so, I get this error:

`Error: Cannot find module '@babel/core'`

Apparently there is no such directory called node_modules/@babel/core. There is only node_modules/@babel/code-frame.

**Comment by gogo9th** [comment]
I tried that. But I get the following error:

`Error: Cannot find module 'babel-preset-es2017' from '/home/skyer/Desktop/expose_runner/chrome-experiment-result'
`

'babel-preset-es2017' is an old package and it should be replaced to @babel/preset-2017. But the program still seems to refer to 'babel-preset-es2017'. I have these installed in package.json:

 ```
 "dependencies": {
    "@babel/preset-es2015": "^7.0.0-beta.53",
    "@babel/preset-es2017": "^7.0.0-beta.53",
    "acorn": "^3.1.0",
    "argparse": "^1.0.10",
    "babel-loader": "^8.1.0",
    "esotope": "^1.4.5",
    "fs": "0.0.1-security",
  },
  "devDependencies": {
    "@babel/cli": "^7.12.1",
    "@babel/core": "^7.12.3",
    "@babel/preset-env": "^7.12.1"
  }
```

**Reply by existentialism** [reply]
Any chance you can post a repo that reproduces the problem?

**Comment by gogo9th** [comment]
I changed the code as follows (the presets part):

```

const babel = require('@babel/core');
const fs = require('fs') 

const code = fs.readFileSync('error.js', 'utf8');
babel.transform(code, {retainLines: true, compact: false, presets: ['@babel/preset-env']}).code;
```

And it works fine. Thanks for your help!
